app/services/teacher_training_public_api/sync_provider.rb

```diff
@@ -17,15 +17,15 @@ def call(run_in_background: true, force_sync_courses: false)
                        end
 
       provider = create_or_update_provider(provider_attrs)
-      sync_courses(run_in_background, provider)
+      sync_courses(run_in_background, provider, incremental_sync: incremental_sync)
     end
 
-    def sync_courses(run_in_background, provider)
+    def sync_courses(run_in_background, provider, incremental_sync: true)
       if sync_courses?
         if run_in_background
-          TeacherTrainingPublicAPI::SyncCourses.perform_async(provider.id, @recruitment_cycle_year)
+          TeacherTrainingPublicAPI::SyncCourses.perform_async(provider.id, @recruitment_cycle_year, incremental_sync)
```

Comment:
I am really not liking this. Sidekiq's `perform_async` method does not accept keyword arguments. is there are better way?

Reply:
I was wondering if a FeatureFlag may be an alternative but then this causes an issue when we want to run either a full sync or incremental sync via the support UI 🤔

Reply:
Yea, thats really annoying that it doesn't accept keyword args 😠 I'm not sure there's a much better way. Maybe someone else will. I don't think it's the end of the world though.